packages/gatsby/src/commands/develop.ts

```diff
@@ -168,7 +168,7 @@ module.exports = async (program: IProgram): Promise<void> => {
   const proxyPort = program.port
   const debugInfo = getDebugInfo(program)
   const [statusServerPort, developPort] = await Promise.all([
-    getRandomPort(),
+    getRandomPort(program.internalStatusPort),
```

Comment:
Rather than a cli option, let's go with just an internal env flag. We can copy paste your description as a comment here.

```suggestion
    // INTERNAL_STATUS_PORT allows for setting the websocket port used for monitoring when the browser should 
    // prompt the user to restart the develop process. This port is randomized by default and in most cases should
    // never be required to configure. It is exposed for environments where port access needs to be explicit, such as
    // with Docker. As the port is meant for internal usage only, any attempt to interface with features it exposes via
    // third-party software is not supported.
    getRandomPort(process.env.INTERNAL_STATUS_PORT),
```

Reply:
You don't want it documented anywhere public? That kinda harms discovery for those that need it? 

The documentation notes for it were rather explicit there is no support for anyone that attempts to interact with the internal API rather than just forwarding the port so that it'd be accessible.

If you do want it documented elsewhere, please let me know which file and I'll place them there as well.